client/app/components/group/_pickupList/pickupList.controller.js

```diff
@@ -191,6 +191,23 @@ class PickupListController {
       }
     });
   }
+
+  joinPickup(pickupId) {
+    let pickupToJoin = this.$filter("filter")(this.allPickups, { id: pickupId });
```

Comment:
The "filter" filter will go away in Angular2+, I wonder if we should avoid using it...